src/chi-vue/src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,338 @@
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { uuid4 } from '../../utils/utils';
+import {
+  activeClass,
+  inverseClass,
+  buttonClass,
+  buttonContentClass,
+  buttonGroupClass,
+  iconButtonClass,
+  lightClass,
+  buttonFlatClass,
+  disabledClass,
+  iconClass,
+  paginationClass,
+  paginationResultsClass,
+  paginationCompactClass,
+  paginationContentClass,
+  paginationLabelClass,
+  paginationPageSizeCLass,
+  paginationStartClass,
+  paginationCenterClass,
+  inputClass,
+  paginationJumperClass,
+  paginationEndClass
+} from '../../constants/classes';
+import { pageChangeEvent, pageSizeChangeEvent } from '../../constants/events';
+
+@Component
+export default class Pagination extends Vue {
+  @Prop() pages!: number;
+  @Prop() firstLast!: boolean;
+  @Prop() inverse!: boolean;
+  @Prop() currentPage!: number;
+  @Prop() results!: number;
+  @Prop() pageSize!: number;
+  @Prop() pageJumper!: number;
+  @Prop() compact!: boolean;
+  @Prop() ariaLabel!: boolean;
+  @Prop() size!: string;
+
+  _pageJumperUuid!: string;
+  _distanceFromCurrent!: number;
+  _lastRenderedPage!: number;
+  _pagesToRender: any = [];
+  _startPage!: JSX.Element | null;
+  _lastPage!: JSX.Element | null;
+  _results: any;
+  _pageSize: any;
+  goToPage: any;
+
+  constructor() {
+    super();
+  }
+
+  calculateDistanceFromCurrent() {
+    switch (this.currentPage) {
+      case 1:
+        this._distanceFromCurrent = 4;
+        break;
+      case 2:
+        this._distanceFromCurrent = 3;
+        break;
+      default:
+        this._distanceFromCurrent = 2;
+    }
+  }
+
+  _goToPage(pageChange: HTMLElement) {
+    let pageToGo;
+    for (
+      let cur = pageChange;
+      cur && !cur.classList.contains(buttonGroupClass);
+      cur = cur.parentNode as HTMLElement
+    ) {
+      if (cur.nodeName === 'BUTTON') {
+        pageToGo = cur.getAttribute('data-page');
+      }
+    }
+    this.$emit(pageChangeEvent, pageToGo);
+  }
+
+  _jumpToPage(jumpTo: string) {
+    this.$emit(pageChangeEvent, jumpTo);
+  }
+
+  _pageSizeChange(size: string) {
+    this.$emit(pageSizeChangeEvent, size);
+  }
+
+  addPage(page: string, icon = '', state = '') {
```

Comment:
Please, add types to the parameters when possible

Reply:
+1

Reply:
I believe giving the default value as `= ''` automatically defines as type string. Am I mistaken?
![Screenshot 2020-10-19 at 23 54 05](https://user-images.githubusercontent.com/25136131/96515891-66777100-1266-11eb-84a0-819e32ac4857.png)

Reply:
No, you are right. It's just a matter of style 😃

Reply:
You're right. Please set type for pageToGo and ariaLabel.